### PR TITLE
optimize: optimize gRPC Interceptor for TCC mode

### PR DESCRIPTION
**Pull request by slievrly**
Signed-off-by: slievrly <slievrly@163.com>

- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did

optimize gRPC Interceptor for TCC mode

### CODE REVIEW

File: integration/grpc/src/main/java/io/seata/integration/grpc/interceptor/server/ServerListenerProxy.java [hunk 1]
```diff
@@ -45,28 +50,38 @@ public void onMessage(ReqT message) {
...
     @Override
     public void onReady() {
         target.onReady();
     }
+
+    private void cleanContext() {
+        if (StringUtils.isNotBlank(xid) && RootContext.inGlobalTransaction()) {
+            RootContext.unbind();
+            BranchType branchType = RootContext.getBranchType();
+            if (BranchType.TCC == branchType) {
```
Comment (a364176773) [inline integration/grpc/src/main/java/io/seata/integration/grpc/interceptor/server/ServerListenerProxy.java:82]: Why not just unbindBranchType?

File: integration/grpc/src/main/java/io/seata/integration/grpc/interceptor/server/ServerListenerProxy.java [hunk 2]
```diff
@@ -15,25 +15,30 @@
...
+import io.seata.core.model.BranchType;
 
 /**
  * @author eddyxu1213@126.com
  */
 public class ServerListenerProxy<ReqT> extends ServerCall.Listener<ReqT> {
 
     private ServerCall.Listener<ReqT> target;
-    private String xid;
+    private final String xid;
```
Comment (a364176773) [inline integration/grpc/src/main/java/io/seata/integration/grpc/interceptor/server/ServerListenerProxy.java:32]: Why is xid not integrated into the context

Comment (slievrly) [reply integration/grpc/src/main/java/io/seata/integration/grpc/interceptor/server/ServerListenerProxy.java:32]: xid is required, context is not required and may be empty.

File: integration/grpc/src/main/java/io/seata/integration/grpc/interceptor/server/ServerListenerProxy.java (hunk 1 above)
Comment (slievrly) [reply integration/grpc/src/main/java/io/seata/integration/grpc/interceptor/server/ServerListenerProxy.java]: done.

**Pull request merged**